@xen-orchestra/lite/src/libs/xen-api.ts

```diff
@@ -278,9 +280,61 @@ export default class XenApi {
 
   async injectWatchEvent(poolRef: string) {
     this.#fromToken = await this.#call("event.inject", [
-      this.sessionId,
+      this.getSessionId(),
       "pool",
       poolRef,
     ]);
   }
+
+  get vm() {
+    return {
+      start: ({ vmsRef }: { vmsRef: string[] }) =>
+        Promise.all(
+          vmsRef.map((vmRef) => this._call("VM.start", [vmRef, false, false]))
+        ),
+      startOn: ({ vmsRef, hostRef }: { vmsRef: string[]; hostRef: string }) =>
+        Promise.all(
+          vmsRef.map((vmRef) =>
+            this._call("VM.start_on", [vmRef, hostRef, false, false])
+          )
+        ),
+      pause: ({ vmsRef }: { vmsRef: string[] }) =>
+        Promise.all(vmsRef.map((vmRef) => this._call("VM.pause", [vmRef]))),
+      suspend: ({ vmsRef }: { vmsRef: string[] }) =>
+        Promise.all(vmsRef.map((vmRef) => this._call("VM.suspend", [vmRef]))),
+      resume: ({
+        vmsRefAndPowerState,
+      }: {
+        vmsRefAndPowerState: { ref: string; powerState: PowerState }[];
+      }) =>
+        Promise.all(
+          vmsRefAndPowerState.map(({ ref, powerState }) => {
+            if (powerState !== "Suspended" && powerState !== "Paused") {
+              console.error("VM.resume: Invalid vm power_state", {
+                $ref: ref,
+                power_state: powerState,
+              });
+              return;
+            }
```

Comment:
The error should be handled when XAPI replies with an error instead of testing a condition before calling the XAPI method.